pkg/services/rpcsrv/server_test.go

```diff
@@ -1856,42 +2050,58 @@ func createValidNotaryRequest(chain *core.Blockchain, sender *keys.PrivateKey, n
 	return p
 }
 
+func runTestCasesWithExecutor(t *testing.T, e *executor, testCases map[string][]rpcTestCase, doRPCCall func(string, string, *testing.T) []byte) {
+	for method, cases := range testCases {
+		t.Run(method, func(t *testing.T) {
+			rpc := `{"jsonrpc": "2.0", "id": 1, "method": "%s", "params": %s}`
+
+			for _, tc := range cases {
+				t.Run(tc.name, func(t *testing.T) {
+					body := doRPCCall(fmt.Sprintf(rpc, method, tc.params), e.httpSrv.URL, t)
+					result := checkErrGetResult(t, body, tc.fail, tc.errCode)
+					if tc.fail {
+						return
+					}
+
+					expected, res := tc.getResultPair(e)
+					err := json.Unmarshal(result, res)
+					require.NoErrorf(t, err, "could not parse response: %s", result)
+
+					if tc.check == nil {
+						assert.Equal(t, expected, res)
+					} else {
+						tc.check(t, e, res)
+					}
+				})
+			}
+		})
+	}
+}
+
 // testRPCProtocol runs a full set of tests using given callback to make actual
 // calls. Some tests change the chain state, thus we reinitialize the chain from
 // scratch here.
 func testRPCProtocol(t *testing.T, doRPCCall func(string, string, *testing.T) []byte) {
+	t.Run("test functions with unsupported states", func(t *testing.T) {
+		chain, rpcSrv, httpSrv := initClearServerWithCustomConfig(t, func(c *config.Config) {
+			c.ApplicationConfiguration.Ledger.KeepOnlyLatestState = true
+		})
+
+		defer chain.Close()
+		defer rpcSrv.Shutdown()
+
+		e := &executor{chain: chain, httpSrv: httpSrv}
+		runTestCasesWithExecutor(t, e, rpcFunctionsWithUnsupportedStatesTestCases, doRPCCall)
+	})
+
 	chain, rpcSrv, httpSrv := initServerWithInMemoryChain(t)
 
 	defer chain.Close()
 	defer rpcSrv.Shutdown()
 
 	e := &executor{chain: chain, httpSrv: httpSrv}
 	t.Run("single request", func(t *testing.T) {
-		for method, cases := range rpcTestCases {
-			t.Run(method, func(t *testing.T) {
-				rpc := `{"jsonrpc": "2.0", "id": 1, "method": "%s", "params": %s}`
-
-				for _, tc := range cases {
-					t.Run(tc.name, func(t *testing.T) {
-						body := doRPCCall(fmt.Sprintf(rpc, method, tc.params), httpSrv.URL, t)
-						result := checkErrGetResult(t, body, tc.fail)
-						if tc.fail {
-							return
-						}
-
-						expected, res := tc.getResultPair(e)
-						err := json.Unmarshal(result, res)
-						require.NoErrorf(t, err, "could not parse response: %s", result)
-
-						if tc.check == nil {
-							assert.Equal(t, expected, res)
-						} else {
-							tc.check(t, e, res)
-						}
-					})
-				}
-			})
-		}
+		runTestCasesWithExecutor(t, e, rpcTestCases, doRPCCall)
 	})
 	t.Run("batch with single request", func(t *testing.T) {
 		for method, cases := range rpcTestCases {
```

Comment:
yes, thanks, done

Reply:
No, it would be hard, because there are multiple requests in a batch. What we can do is to add an unused paratemeter to `checkErrGetBatchResult` so that its signature matches the `checkErrGetResult` and after that pass one of the checking function to the `runTestCasesWithExecutor` as a parameter.

Reply:
done